nanoFramework.DependencyInjection/System/IServiceProviderIsService.cs

```diff
@@ -0,0 +1,15 @@
+﻿namespace System
```

Comment:
Is this the correct namespace? I put it here because that is where IServiceProvider was but I also see IServiceCollection under nanoFramework.DependencyInjection (which I think might also be the wrong namespace if we are following .NET official)

Reply:
I think it should either be `Microsoft.Extensions.DependencyInjection` or `nanoFramework.DependencyInjection` but not `System`.

Reply:
I think `Microsoft.Extensions.DependencyInjection` makes the most sense if the goal it to be as close to the official API as possible.

If we're going with a breaking change I would prefer to break it all and move everything to the correct namespace (either solution is breaking since `IServiceProvider` is already in `System` and at least should be moved to `nanoFramework.DependencyInjection`)

Reply:
Let's stay away from `System` here.
I agree with `Microsoft.Extensions` for cross platform compatibility.

Reply:
> FYI - The IServiceProvider interface is part of the System namespace in the full version of .NET today.
> 
> https://learn.microsoft.com/en-us/dotnet/api/system.iserviceprovider?view=net-7.0

Thanks for checking that.

I looked up all the classes and updated them to be in the same namespace as the full version of .NET.

This is now a breaking change.